### PR TITLE
Refactor chunk, metadata IDs

### PR DESCRIPTION
**Pull request by bryanlb**
Fixes consistency issues, as well as collisions issues on chunk IDs.

### CODE REVIEW

File: kaldb/src/main/java/com/slack/kaldb/metadata/snapshot/SnapshotMetadataStore.java [hunk 1]
```diff
@@ -6,7 +6,7 @@
 import org.slf4j.LoggerFactory;
 
 public class SnapshotMetadataStore extends PersistentMutableMetadataStore<SnapshotMetadata> {
-  public static final String SNAPSHOT_METADATA_STORE_ZK_PATH = "/snapshots";
+  public static final String SNAPSHOT_METADATA_STORE_ZK_PATH = "/snapshot";
```
Comment (bryanlb) [inline kaldb/src/main/java/com/slack/kaldb/metadata/snapshot/SnapshotMetadataStore.java:9]: This is the only ZK path that is plural, the rest are singular (search, replica, cacheSlot, etc)

File: kaldb/src/main/java/com/slack/kaldb/chunk/ReadOnlyChunkImpl.java [hunk 2]
```diff
@@ -155,10 +155,8 @@ private KaldbMetadataStoreChangeListener cacheNodeListener() {
 
   private void registerSearchMetadata(String snapshotName)
       throws ExecutionException, InterruptedException, TimeoutException {
-    // for "name" we pass url so that we can run two JVMs on the same host and not have znode path
-    // collision.
     this.searchMetadata =
-        new SearchMetadata(searchContext.toName(), snapshotName, searchContext.toUrl());
+        new SearchMetadata(snapshotName + "_" + searchContext.hostname, snapshotName, searchContext.toUrl());
```
Comment (bryanlb) [inline kaldb/src/main/java/com/slack/kaldb/chunk/ReadOnlyChunkImpl.java:159]: Prevents collisions when a single node needs to register multiple search metadata (ie, cache nodes)

File: kaldb/src/main/java/com/slack/kaldb/chunk/ReadWriteChunk.java [hunk 3]
```diff
@@ -121,7 +123,7 @@ protected ReadWriteChunk(
   public abstract void preClose();
 
   private SearchMetadata toSearchMetadata(String snapshotName, SearchContext searchContext) {
-    return new SearchMetadata(snapshotName, snapshotName, searchContext.toUrl());
+    return new SearchMetadata(snapshotName + "_" + searchContext.hostname, snapshotName, searchContext.toUrl());
```
Comment (bryanlb) [inline kaldb/src/main/java/com/slack/kaldb/chunk/ReadWriteChunk.java:126]: Adds hostname for easier debugging and consistency with the ReadOnlyChunk

File: kaldb/src/main/java/com/slack/kaldb/chunk/ReadOnlyChunkImpl.java [hunk 4]
```diff
@@ -155,10 +155,9 @@ private KaldbMetadataStoreChangeListener cacheNodeListener() {
 
   private void registerSearchMetadata(String snapshotName)
       throws ExecutionException, InterruptedException, TimeoutException {
-    // for "name" we pass url so that we can run two JVMs on the same host and not have znode path
-    // collision.
     this.searchMetadata =
-        new SearchMetadata(searchContext.toName(), snapshotName, searchContext.toUrl());
+        new SearchMetadata(
+            snapshotName + "_" + searchContext.hostname, snapshotName, searchContext.toUrl());
```
Comment (mansu) [inline kaldb/src/main/java/com/slack/kaldb/chunk/ReadOnlyChunkImpl.java:160]: I think it's best to update the toName() method in SearchContext or add a new method in SearchMetadata to keep all the name building logic in the same place. This logic is already repeated twice here. 

May be make it static so we can use it in unit tests also.

File: kaldb/src/main/java/com/slack/kaldb/metadata/snapshot/SnapshotMetadataStore.java (hunk 1 above)
Comment (mansu) [reply kaldb/src/main/java/com/slack/kaldb/metadata/snapshot/SnapshotMetadataStore.java:9]: +1 for consistency.

File: kaldb/src/test/java/com/slack/kaldb/chunk/ReadOnlyChunkImplTest.java [hunk 5]
```diff
@@ -153,7 +153,7 @@ public void shouldHandleChunkLivecycle() throws Exception {
     assertThat(searchMetadataStore.getCached().get(0).snapshotName).isEqualTo(snapshotId);
     assertThat(searchMetadataStore.getCached().get(0).url)
         .isEqualTo("gproto+http://localhost:8080");
-    assertThat(searchMetadataStore.getCached().get(0).name).isEqualTo("localhost:8080");
+    assertThat(searchMetadataStore.getCached().get(0).name).isEqualTo(snapshotId + "_localhost");
```
Comment (mansu) [inline kaldb/src/test/java/com/slack/kaldb/chunk/ReadOnlyChunkImplTest.java:156]: I think it's best to call the toName method here to check the name.

**Review by mansu — CHANGES_REQUESTED** [review]

File: kaldb/src/main/java/com/slack/kaldb/chunk/ReadOnlyChunkImpl.java (hunk 4 above)
Comment (bryanlb) [reply kaldb/src/main/java/com/slack/kaldb/chunk/ReadOnlyChunkImpl.java]: Done.

File: kaldb/src/test/java/com/slack/kaldb/chunk/ReadOnlyChunkImplTest.java (hunk 5 above)
Comment (bryanlb) [reply kaldb/src/test/java/com/slack/kaldb/chunk/ReadOnlyChunkImplTest.java]: Done.

**Review by mansu — APPROVED** [review]

**Pull request merged**